test/integration/buffer_accounting_integration_test.cc

```diff
@@ -358,6 +366,61 @@ TEST_P(Http2BufferWatermarksTest, ShouldTrackAllocatedBytesToUpstream) {
   }
 }
 
+TEST_P(Http2BufferWatermarksTest, ShouldTrackAllocatedBytesToShadowUpstream) {
+  const int num_requests = 5;
+  const uint32_t request_body_size = 4096;
+  const uint32_t response_body_size = 4096;
+  TestScopedRuntime scoped_runtime;
+  scoped_runtime.mergeValues({{"envoy.reloadable_features.streaming_shadow", "true"}});
+
+  autonomous_upstream_ = true;
+  autonomous_allow_incomplete_streams_ = true;
+  setUpstreamCount(2);
+  config_helper_.addConfigModifier([](envoy::config::bootstrap::v3::Bootstrap& bootstrap) {
+    auto* cluster = bootstrap.mutable_static_resources()->add_clusters();
+    cluster->MergeFrom(bootstrap.static_resources().clusters()[0]);
+    cluster->set_name("cluster_1");
+  });
+  config_helper_.addConfigModifier(
+      [=](envoy::extensions::filters::network::http_connection_manager::v3::HttpConnectionManager&
+              hcm) -> void {
+        auto* mirror_policy = hcm.mutable_route_config()
+                                  ->mutable_virtual_hosts(0)
+                                  ->mutable_routes(0)
+                                  ->mutable_route()
+                                  ->add_request_mirror_policies();
+        mirror_policy->set_cluster("cluster_1");
+      });
+  initialize();
+
+  buffer_factory_->setExpectedAccountBalance(request_body_size, num_requests);
```

Comment:
should the account balance be 2x? Since it's many copied to the original and the mirror

Reply:
Done.